client/blocks/comment-button/index.jsx

```diff
@@ -39,7 +45,7 @@ const CommentButton = React.createClass( {
 	render() {
 		let label;
 		const containerTag = this.props.tagName,
-			commentCount = this.props.commentCount;
+			commentCount = ( this.props.syncedCount === undefined ? this.props.commentCount : this.props.syncedCount );
```

Comment:
> In the future, when we're using redux to store post state, we can populate the redux state.comments.totalCommentCount, and remove the commentCount prop

We are using redux to hold posts state already, it's just not used yet for the main post-cards in `/posts` (but it is used for the draft items in the sidebar).

If you can't rely on one selector for getting the comment count, I'd probably rename the props to have an `initialCount` or `defaultCount` for passing the value down from the post. The data read from state should be the canonical, and thus match the main prop. (Which, by the way, could be just `count`.)